core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -451,6 +465,32 @@ class DAGScheduler(
     parents
   }
 
+  /**
+   * Traverse all the parent RDDs within the same stage with the given RDD, check whether all the
```

Comment:
It also checks the RDD itself. "Traverses the given RDD and its ancestors within the same stage and checks whether all of the RDDs satisfy a given predicate."